ckanext/datastore/blueprint.py

```diff
@@ -136,6 +138,12 @@ class DictionaryView(MethodView):
 
     def _prepare(self, id: str, resource_id: str) -> dict[str, Any]:
         try:
+            check_access(
+                "datastore_upsert",
+                context={"user": current_user.name, "auth_user_obj": current_user},
+                data_dict={"id": id, "resource_id": resource_id},
```

Comment:
We should use the correct action that will be called
```suggestion
                "datastore_create",
                context={"user": current_user.name, "auth_user_obj": current_user},
                data_dict={"resource_id": resource_id},
```

Reply:
you're right